### ISSUE TITLE
Vagrant Creates Two Extra Network Interfaces... Always.

### ISSUE THREAD

**Issue by ericpulvino** [mitchellh/vagrant#7286]
### Vagrant version
Vagrant 1.8.1

### Host operating system
Ubuntu Xenial 16.04

### Guest operating system
CumulusCommunity/cumulus-vx
boxcutter/ubuntu1404

### Vagrantfile
```ruby
# Created by Topology-Converter v4.0.4
#    using topology data from: ./topology.dot
#    NOTE: in order to use this Vagrantfile you will need:
#       -Vagrant(v1.7+) installed: http://www.vagrantup.com/downloads 
#       -Cumulus Plugin for Vagrant installed: $ vagrant plugin install vagrant-cumulus
#       -the "helper_scripts" directory that comes packaged with topology-converter.py 
#       -Virtualbox installed: https://www.virtualbox.org/wiki/Downloads

raise "vagrant-cumulus plugin must be installed, try $ vagrant plugin install vagrant-cumulus" unless Vagrant.has_plugin? "vagrant-cumulus"

Vagrant.configure("2") do |config|
  wbid = 1462334090

  config.vm.provider "virtualbox" do |v|
    v.gui=false

  end


  ##### DEFINE VM for leaf02 #####
  config.vm.define "leaf02" do |device|
    device.vm.hostname = "leaf02"
    device.vm.box = "CumulusCommunity/cumulus-vx"
    device.vm.provider "virtualbox" do |v|
      v.name = "1462334090_leaf02"
      v.memory = 300
    end
  config.vm.synced_folder ".", "/vagrant", disabled: true

      # link for swp49 --> leaf01:swp52
      device.vm.network "private_network", virtualbox__intnet: "{wbid}_net1", auto_config: false , :mac => "443839000023"

    device.vm.provider "virtualbox" do |vbox|
      vbox.customize ['modifyvm', :id, '--nicpromisc2', 'allow-vms']

      vbox.customize ["modifyvm", :id, "--nictype1", "virtio"]
      vbox.customize ["modifyvm", :id, "--nictype2", "virtio"]

    end

      # Run Any Extra Config
      device.vm.provision :shell , path: "./helper_scripts/extra_switch_config.sh"

      # Apply the interface re-map
        #Disable default remap on Cumulus VX 
      device.vm.provision :shell , inline: "mv /etc/init.d/rename_eth_swp /etc/init.d/rename_eth_swp.old"
      device.vm.provision "file", source: "./helper_scripts/apply_udev.py", destination: "/home/vagrant/apply_udev.py"
      device.vm.provision :shell , inline: "chmod 755 /home/vagrant/apply_udev.py"
      device.vm.provision :shell , inline: "/home/vagrant/apply_udev.py -a 44:38:39:00:00:23 swp49"

      device.vm.provision :shell , inline: "/home/vagrant/apply_udev.py -vm"
      device.vm.provision :shell , inline: "/home/vagrant/apply_udev.py -s"
      device.vm.provision :shell , inline: "reboot"

  end

  ##### DEFINE VM for leaf01 #####
  config.vm.define "leaf01" do |device|
    device.vm.hostname = "leaf01"
    device.vm.box = "CumulusCommunity/cumulus-vx"
    device.vm.provider "virtualbox" do |v|
      v.name = "1462334090_leaf01"
      v.memory = 300
    end
  config.vm.synced_folder ".", "/vagrant", disabled: true

      # link for swp2 --> leaf03:swp2
      device.vm.network "private_network", virtualbox__intnet: "{wbid}_net2", auto_config: false , :mac => "443839000001"
      
      # link for swp3 --> leaf03:swp3
      device.vm.network "private_network", virtualbox__intnet: "{wbid}_net4", auto_config: false , :mac => "443839000005"
      
      # link for swp1 --> leaf03:swp1
      device.vm.network "private_network", virtualbox__intnet: "{wbid}_net3", auto_config: false , :mac => "443839000003"
      
      # link for swp4 --> leaf03:swp4
      device.vm.network "private_network", virtualbox__intnet: "{wbid}_net5", auto_config: false , :mac => "443839000007"
      
      # link for swp52 --> leaf02:swp49
      device.vm.network "private_network", virtualbox__intnet: "{wbid}_net1", auto_config: false , :mac => "443839000024"

    device.vm.provider "virtualbox" do |vbox|
      vbox.customize ['modifyvm', :id, '--nicpromisc2', 'allow-vms']
      vbox.customize ['modifyvm', :id, '--nicpromisc3', 'allow-vms']
      vbox.customize ['modifyvm', :id, '--nicpromisc4', 'allow-vms']
      vbox.customize ['modifyvm', :id, '--nicpromisc5', 'allow-vms']
      vbox.customize ['modifyvm', :id, '--nicpromisc6', 'allow-vms']

      vbox.customize ["modifyvm", :id, "--nictype1", "virtio"]
      vbox.customize ["modifyvm", :id, "--nictype2", "virtio"]
      vbox.customize ["modifyvm", :id, "--nictype3", "virtio"]
      vbox.customize ["modifyvm", :id, "--nictype4", "virtio"]
      vbox.customize ["modifyvm", :id, "--nictype5", "virtio"]
      vbox.customize ["modifyvm", :id, "--nictype6", "virtio"]

    end

      # Run Any Extra Config
      device.vm.provision :shell , path: "./helper_scripts/extra_switch_config.sh"


      # Apply the interface re-map
        #Disable default remap on Cumulus VX 
      device.vm.provision :shell , inline: "mv /etc/init.d/rename_eth_swp /etc/init.d/rename_eth_swp.old"
      device.vm.provision "file", source: "./helper_scripts/apply_udev.py", destination: "/home/vagrant/apply_udev.py"
      device.vm.provision :shell , inline: "chmod 755 /home/vagrant/apply_udev.py"
      device.vm.provision :shell , inline: "/home/vagrant/apply_udev.py -a 443839000001 swp2"
      device.vm.provision :shell , inline: "/home/vagrant/apply_udev.py -a 443839000005 swp3"
      device.vm.provision :shell , inline: "/home/vagrant/apply_udev.py -a 443839000003 swp1"
      device.vm.provision :shell , inline: "/home/vagrant/apply_udev.py -a 443839000007 swp4"
      device.vm.provision :shell , inline: "/home/vagrant/apply_udev.py -a 44:38:39:00:00:24 swp52"

      device.vm.provision :shell , inline: "/home/vagrant/apply_udev.py -vm"
      device.vm.provision :shell , inline: "/home/vagrant/apply_udev.py -s"
      device.vm.provision :shell , inline: "reboot"

  end
end
```

### Debug output
https://gist.github.com/ericpulvino/cda44abcf7a49479d7de24dab32667d8

### Expected behavior
There should be 3 interfaces on Leaf02. 
(Like this....)

vagrant@leaf02:~$ ip link show
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 16436 qdisc noqueue state UNKNOWN mode DEFAULT 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: vagrant: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP mode DEFAULT qlen 1000
    link/ether 08:00:27:d2:8f:2a brd ff:ff:ff:ff:ff:ff
3: swp49: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT qlen 1000
    link/ether 44:38:39:00:00:23 brd ff:ff:ff:ff:ff:ff

### Actual behavior
Instead there are 5... two extra seem to be created at the end.

vagrant@leaf02:~$ ip link show
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 16436 qdisc noqueue state UNKNOWN mode DEFAULT 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: vagrant: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP mode DEFAULT qlen 1000
    link/ether 08:00:27:d2:8f:2a brd ff:ff:ff:ff:ff:ff
3: swp49: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT qlen 1000
    link/ether 44:38:39:00:00:23 brd ff:ff:ff:ff:ff:ff
**4: eth2: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT qlen 1000
    link/ether 08:00:27:88:13:44 brd ff:ff:ff:ff:ff:ff
5: eth3: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT qlen 1000
    link/ether 08:00:27:4a:5a:99 brd ff:ff:ff:ff:ff:ff**

### Steps to reproduce
Seems to work in all the examples I play with on all OSes... is this a vagrantfile syntax issue perhaps?
1. vagrant up
2. vagrant ssh leaf02
3. ip link show

**Comment by ericpulvino** [comment]
I've root caused this issue...

"provider.driver.max_network_adapters" is called in line 15 of this file -->
https://github.com/mitchellh/vagrant/blob/1f88a9737369eda3dede0d3aeb30928adf82a9f7/plugins/providers/virtualbox/action/clear_network_interfaces.rb

`        def call(env)
          # Create the adapters array to make all adapters nothing.
          # **We do adapters 2 to 8 because that is every built-in adapter**
          # excluding the NAT adapter on port 1 which Vagrant always
          # expects to exist.
          adapters = []
          2.upto(env[:machine].provider.driver.max_network_adapters).each do |i|
            adapters << {
              adapter: i,
              type:    :none
            }
          end`

Unfortunately this approach does not encompass every adapter. Only adapters 2-8 are being cleared... in reality Virtualbox ships with ~36 adapters, 1 being reserved for the vagrant interface. So a better method would be to clear all adapters that exists minus the Vagrant interface (or adapters 2-36).

It looks like "max_network_adapters" is defined as "8" in line 181 of this file... it should be 36 for virtualbox. --> https://github.com/mitchellh/vagrant/blob/1f88a9737369eda3dede0d3aeb30928adf82a9f7/plugins/providers/virtualbox/driver/base.rb

Submitting a pull request with this modification.

**Comment by sethvargo** [comment]
Fixed in #7293
